crates/ruff_python_parser/tests/snapshots/invalid_syntax@re_lex_logical_token.py.snap

```diff
@@ -0,0 +1,498 @@
+                                            ],
+                                            ctx: Load,
+                                        },
+                                    ),
+                                ],
+                                keywords: [],
+                            },
+                        },
+                    ),
+                    body: [
+                        FunctionDef(
+                            StmtFunctionDef {
+                                range: 588..611,
+                                is_async: false,
+                                decorator_list: [],
+                                name: Identifier {
+                                    id: "bar",
+                                    range: 592..595,
+                                },
+                                type_params: None,
+                                parameters: Parameters {
+                                    range: 595..597,
+                                    posonlyargs: [],
+                                    args: [],
+                                    vararg: None,
+                                    kwonlyargs: [],
+                                    kwarg: None,
+                                },
+                                returns: None,
+                                body: [
+                                    Pass(
+                                        StmtPass {
+                                            range: 607..611,
+                                        },
+                                    ),
+                                ],
+                            },
+                        ),
+                    ],
+                    elif_else_clauses: [],
+                },
+            ),
+        ],
+    },
+)
+```
+## Errors
+
+  |
+1 | # No indentation before the function definition
+2 | if call(foo
+  |            ^ Syntax Error: Expected ')', found newline
+3 | def bar():
+4 |     pass
+  |
+
+
+  |
+1 | # No indentation before the function definition
+2 | if call(foo
+3 | def bar():
+  | ^^^ Syntax Error: Expected an indented block after `if` statement
+4 |     pass
+  |
+
+
+   |
+ 7 | # Indented function definition
+ 8 | if call(foo
+   |            ^ Syntax Error: Expected ')', found newline
+ 9 |     def bar():
+10 |         pass
+   |
+
+
+   |
+13 | # There are multiple non-logical newlines (blank lines) in the `if` body
+14 | if call(foo
+   |            ^ Syntax Error: Expected ')', found newline
+15 | 
+16 | 
+17 |     def bar():
+   |
+
+
+   |
+21 | # There are trailing whitespaces in the blank line inside the `if` body
+22 | if call(foo
+   |            ^ Syntax Error: Expected ')', found newline
+23 |         
+24 |     def bar():
+25 |         pass
+   |
+
+
+   |
+28 | # The lexer is nested with multiple levels of parentheses
+29 | if call(foo, [a, b
+   |                   ^ Syntax Error: Expected ']', found NonLogicalNewline
```

Comment:
Should we map `NonLogicalNewline` to newline? It seems an odd error message to show to users.

Reply:
Yeah, it is odd. I could map it to "newline" in the `Display` implementation.